src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1567,9 +1567,10 @@ public static bool IsLeapYear(int year)
             {
```

Comment:
If we're trying to optimize this method, do we also need to optimize this check? Will the JIT already emit code as good as it does for `(uint)(year - 1) > 9998`?

Reply:
@stephentoub yep, jit does it (https://github.com/dotnet/runtime/pull/93521)